### ISSUE TITLE
"er.a is not a valid duofern device" since v0.4.3

### ISSUE THREAD

**Issue by bcutter** [gluap/pyduofern-hacs#31]
Probably when using `duofern.ask_for_update` service (but that message has only been logged once, even I used the service without an device_id few more times):

```
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.duofern
Source: custom_components/duofern/__init__.py:113
Integration: Duofern control (documentation, issues)
First occurred: 01:45:00 (1 occurrences)
Last logged: 01:45:00

er.a is not a valid duofern device, I only know dict_keys(['xxxxxx', 'yyyyyy', 'zzzzzz', 'aaaaaa', 'bbbbbb', 'cccccc', 'dddddd', 'eeeeee', '000000', 'ffffff'])
```

Related to https://github.com/gluap/pyduofern-hacs/releases/tag/v0.4.1:
- The `device_id` parameter of `duofern.ask_for_update` is mandatory now, isn't it?
- So applying this to all devices or (important to me) using a cover group (entity `cover.all_covers`) is not possible anymore?
_That would make that service pretty useless, e. g. when HA starts and I want to sync the device states (without needing to wait for 5 minutes)..._ 😞 It was working perfectly for all my devices. **Now we don't have an option to IMMEDIATELY sync device states, or am I missing something?**

![grafik](https://user-images.githubusercontent.com/13799156/226773683-835df07d-c206-44f6-8e5b-5a3810abfc23.png)

**Comment by gluap** [comment]
Tbh that wasn't possible in the past either - if anything, it failed silently. 

Making the update device specific was requested by you because "updating all" was not granular enough. So I put in the work to add it and removed the "update all" part as you, the only user then, had deemed it unsuitable.

Calling update_device_state for a device group has the same downsides as calling it for all devices - messages sent out in a burst. So I didn't even think that would be what you meant.

**Comment by bcutter** [comment]
So what do you recommend for syncing the device state for all devices immediately?

**Comment by gluap** [comment]
I have no proposal for that. My takeaway from https://github.com/gluap/pyduofern-hacs/issues/24#issuecomment-1407490248 was "updating individual devices is preferrable".

**Comment by bcutter** [comment]
I still think both is needed and useful.

Specifying an entity_id is helpful and common HA standard (usually with target: entity_id) for services.

Applying the service to all entity_ids too (cited from #24):

> Anyway it would still be good if it would work with groups, so setting the "entity_id" to a "cover.group_of_covers" would be necessary from my point of view.

I did not try it yet if the service accepts magic (hidden/not well or at all documented) keys like e. g. lights services do with "*". At least the GUI according to services.yaml only lists/accepts actual device entity_id's.

My summary: the current implementation (0.4.3) with targeting single entity_id's is very good.

Aiming at forcing an immediate device status update for ALL devices it would be great if the service accepts groups or simply addresses all devices if no entity_id is specified (like the former service prior to renaming it did).

As (ugly) not very end user friendly workarounds I can currently only imagine
- using the service for every entity_id so for 10 devices that makes 10 service calls
- trying to template service usage e. g. in a for loop

**Comment by gluap** [comment]
I can check if i can merge back the "update all" solution, and / or if there's a way to make the gui more intuitive for specifying lists of targets or a group. I think I have seen that for other services so probably it's possible.

Will take until the weekend though, as im traveling without my laptop.

**Comment by bcutter** [comment]
I will test the functionality of the "multiple entities" and "all entities" tricks.

If it works in my opinion there's no need for (investing time on) a rollback of the former "update all" handling. Just a docs update (once that version leaves beta status), 5 minutes, done.

Of course using the UI would be much more intuitive for the "apply for multiple entities" part, but tbh I don't think one is really (regularly) using this service from the UI. Maybe I'm just projecting MY usage to others here (biased opinion). But:

Services are used mainly as part of scripts or automations so once one knows how to syntactically use the service for multiple/all entities, it should be fine.

**Comment by bcutter** [comment]
Update: again one step back. 😞 

Using
```
service: duofern.ask_for_update
data:
  device_id: all
```

as well as
```
service: duofern.ask_for_update
data:
  device_id: cover.cover_a,cover.cover_b,cover.cover_c
```

gives the well-known error message in HA log:

❌ all (LOG truncated):

```
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.duofern
Source: custom_components/duofern/__init__.py:113
Integration: Duofern control (documentation, issues)
First occurred: 01:45:00 (10 occurrences)
Last logged: 22:43:31

all is not a valid duofern device, I only know dict_keys(['...']...)
...
```

❌ multiple entities specified (LOG truncated):
```
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: custom_components.duofern
Source: custom_components/duofern/__init__.py:113
Integration: Duofern control (documentation, issues)
First occurred: 01:45:00 (13 occurrences)
Last logged: 22:48:43

er.er.a is not a valid duofern device, I only know dict_keys(['...']...)
...
```

So even it looks totally fine in the UI, that error is logged - and it's very likely not working on duoFERN side.
Probably duoFERN simply can neither map "all" to one of the expected device_ids nor work with more than one device_id.

---

So I fear: yes, unfortunately
> I can check if i can merge back the "update all" solution

...seems to be necessary to bring this back to work.

**Comment by gluap** [comment]
thank you for testing this so extensively. I'm looking into it!

**Comment by bcutter** [comment]
You're welcome. Thanks four having a look into it, really looking forward to that. For now without the multiple/all `duofern.ask_for_update` service I'm running into the need for "5 minutes auto update" function, which is quite a lot. With the service I can reduce the status not updating again to only few seconds instead of 5 minutes without the need to rely on oldschool `cover.stop_cover` service.

![grafik](https://user-images.githubusercontent.com/13799156/227794479-c7985c7e-b635-4fa7-a5b5-ddbe6907af9f.png)

**Comment by bcutter** [comment]
Seems to be fixed with duoFERN v0.5.5 where among others using the `all: true` feature is working fine now. Closed for me right now.
